src/compiler/checker.ts

```diff
@@ -9530,7 +9530,7 @@ namespace ts {
              * * Ternary.Maybe if they are related with assumptions of other relationships, or
              * * Ternary.False if they are not related.
              */
-            function isRelatedTo(source: Type, target: Type, reportErrors?: boolean, headMessage?: DiagnosticMessage): Ternary {
+            function isRelatedTo(source: Type, target: Type, reportErrors?: boolean, headMessage?: DiagnosticMessage, dontReportMatchedConstructor = false): Ternary {

```

Comment:
I don't know exactly where you'd instead put it, but adding this change to `isRelatedTo` is probably too broad of an API to account for the change